pkg/internal/tests/cross-tests/diff.go

```diff
@@ -0,0 +1,39 @@
+package crosstests
+
+import (
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/zclconf/go-cty/cty"
+
+	crosstestsimpl "github.com/pulumi/pulumi-terraform-bridge/v3/pkg/internal/tests/cross-tests/impl"
+)
+
+type diffOpts struct {
+	deleteBeforeReplace bool
+}
+
+// An option that can be used to customize [Create].
+type DiffOption func(*diffOpts)
+
+// DiffDeleteBeforeReplace specifies a timeout option for [Create].
+func DiffDeleteBeforeReplace(deleteBeforeReplace bool) DiffOption {
+	return func(o *diffOpts) { o.deleteBeforeReplace = deleteBeforeReplace }
+}
+
+func Diff(
+	t T, resource *schema.Resource, config1, config2 map[string]cty.Value, opts ...DiffOption,
```

Comment:
Yeah, this unfortunately limits the expressiveness of the test too - for example custom Create and Update methods might be required in a Diff test for implementing Computed properties. Instead of re-implementing the API with options we should have a simple check for the banned inputs.

Reply:
I don't see a good reason why we would choose to add a new interface there for specifying schemas instead of reusing the default TF SDK one - it works well enough and allows us to copy-paste provider examples and avoids unnecessary complexity.

I can add refactoring the other cross-test interfaces to my TODO list

Reply:
I'm on the side of being more permissive here in the internal test code.

Reply:
> I don't see a good reason why we would choose to add a new interface there for specifying schemas instead of reusing the default TF SDK one

Because it keeps the tests intentional. We only add properties that can actually be tested. 

>  it works well enough and allows us to copy-paste provider examples and avoids unnecessary complexity.

I'd rather we didn't copy-paste whole resources to test only a small part of them.

> I don't see what Resource.Importer has to do with Diff test

Nothing at all, but I can set `.Importer` on a `*schema.Resource` and pass that to `crosstests.Diff` with this design. I'd rather that was a compiler error. I use that example because I remember someone setting `.Importer` for a create and asserting that the `.Importer` didn't panic.

I won't block the PR on this, but IMO passing `*schema.Resource` directly makes the tests easier to write but harder to read.

Reply:
>passing *schema.Resource directly makes the tests easier to write but harder to read.

I don't agree about readability either. I think the benefit of options is that it is harder to get the Schema wrong. Given that there's very few people who actually write these type of tests, I think it's worth keeping it simple for now and evolving the interface once we find a need to.

Let's revisit once we have a concrete problem we are solving with the extra interface.